doc/technotes/errorHandling.rst

```diff
@@ -0,0 +1,190 @@
+.. _readme-errorHandling:
+
+==============
+Error Handling
+==============
+
+This README describes the error handling feature as currently implemented.
+
+
+Errors
+------
+A base class ``Error`` is defined in the Chapel standard modules.
+
+::
+
+  class Error {
+    var msg: string;
+
+    proc Error(_msg: string) {
+      msg = _msg;
+    }
+  }
+
+Users may use the provided ``Error`` directly, or create their own error
+hierarchy with ``Error`` at its root.
+
+::
+
+  class MyError: Error {}
+
+  class MyIntError: MyError {
+    var i: int;
+  }
+
+Chapel does not provide a standard set of ``Error`` classes.
+
+
+Throwing Errors
+---------------
+``throw`` statements throw errors from a function to its callee.
+
+::
+
+  throw new Error("with a message");
+
+  var e: Error;
+  // ... set e ...
+  throw e;
+
+If a function may throw an error, it must be declared with ``throws``.
+
+::
+
+  proc mayThrowErrors() throws { ... }
+  proc mayThrowErrorsAlso(): A throws where ... { ... }
+  proc mayNotThrowErrors() { ... }
+
+
+Handling Errors
+---------------
+
+Default Policy
```

Comment:
By the time I got to the end, I figured out what you meant by "Default Policy". I think you need a paragraph above these sections to introduce the outline of the following. e.g.

Error handling behavior can be broken up into several cases. Because it is possible to choose between two modes, default mode and strict mode, we describe the behaviors separately: